examples/templates/azure-linux/main.tf

```diff
@@ -11,64 +11,206 @@ terraform {
+  }
+  option {
+    name  = "Korea (Seoul)"
+    value = "koreacentral"
+    icon  = "/emojis/1f1f0-1f1f7.png"
+  }
+  option {
+    name  = "Europe (Ireland)"
+    value = "northeurope"
+    icon  = "/emojis/1f1ea-1f1fa.png"
+  }
+  option {
+    name  = "Europe (Netherlands)"
+    value = "westeurope"
+    icon  = "/emojis/1f1ea-1f1fa.png"
+  }
+  option {
+    name  = "France (Paris)"
+    value = "francecentral"
+    icon  = "/emojis/1f1eb-1f1f7.png"
+  }
+  option {
+    name  = "Germany (Frankfurt)"
+    value = "germanywestcentral"
+    icon  = "/emojis/1f1e9-1f1ea.png"
+  }
+  option {
+    name  = "Norway (Oslo)"
+    value = "norwayeast"
+    icon  = "/emojis/1f1f3-1f1f4.png"
+  }
+  option {
+    name  = "Sweden (Gävle)"
+    value = "swedencentral"
+    icon  = "/emojis/1f1f8-1f1ea.png"
+  }
+  option {
+    name  = "Switzerland (Zurich)"
+    value = "switzerlandnorth"
+    icon  = "/emojis/1f1e8-1f1ed.png"
+  }
+  option {
+    name  = "Qatar (Doha)"
+    value = "qatarcentral"
+    icon  = "/emojis/1f1f6-1f1e6.png"
+  }
+  option {
+    name  = "UAE (Dubai)"
+    value = "uaenorth"
+    icon  = "/emojis/1f1e6-1f1ea.png"
+  }
+  option {
+    name  = "South Africa (Johannesburg)"
+    value = "southafricanorth"
+    icon  = "/emojis/1f1ff-1f1e6.png"
+  }
+  option {
+    name  = "UK (London)"
+    value = "uksouth"
+    icon  = "/emojis/1f1ec-1f1e7.png"
+  }
+}
+
+data "coder_parameter" "instance_type" {
+  name        = "Instance Type"
   description = "What instance type should your workspace use?"
   default     = "Standard_B4ms"
-  validation {
-    condition = contains([
-      "Standard_B1ms",
-      "Standard_B2ms",
-      "Standard_B4ms",
-      "Standard_B8ms",
-      "Standard_B12ms",
-      "Standard_B16ms",
-      "Standard_D2as_v5",
-      "Standard_D4as_v5",
-      "Standard_D8as_v5",
-      "Standard_D16as_v5",
-      "Standard_D32as_v5",
-    ], var.instance_type)
-    error_message = "Invalid instance type!"
-  }
-}
-
-variable "home_size" {
-  type        = number
+  icon        = "/icons/azure.png"
+  mutable     = false
+  option {
+    name  = "1 vCPU, 2 GiB RAM"
+    value = "Standard_B1ms"
+  }
+  option {
+    name  = "2 vCPU, 8 GiB RAM"
+    value = "Standard_B2ms"
+  }
+  option {
+    name  = "4 vCPU, 16 GiB RAM"
+    value = "Standard_B4ms"
+  }
+  option {
+    name  = "8 vCPU, 32 GiB RAM"
+    value = "Standard_B8ms"
+  }
+  option {
+    name  = "12 vCPU, 48 GiB RAM"
+    value = "Standard_B12ms"
+  }
+  option {
+    name  = "16 vCPU, 64 GiB RAM"
+    value = "Standard_B16ms"
+  }
+  option {
+    name  = "2 vCPU, 8 GiB RAM"
+    value = "Standard_D2as_v5"
+  }
+  option {
+    name  = "4 vCPU, 16 GiB RAM"
+    value = "Standard_D4as_v5"
+  }
+  option {
+    name  = "8 vCPU, 32 GiB RAM"
+    value = "Standard_D8as_v5"
+  }
+  option {
+    name  = "16 vCPU, 64 GiB RAM"
+    value = "Standard_D16as_v5"
+  }
+  option {
+    name  = "32 vCPU, 128 GiB RAM"
+    value = "Standard_D32as_v5"
+  }
```

Comment:
I'm afraid that we have to stick to the setup names like "Standard D32 as ...". Otherwise:

```
Error: multiple options cannot have the same name "2 vCPU, 8 GiB RAM"
Error: building AzureRM Client: please ensure you have installed Azure CLI version 2.0.79 or newer. Error parsing json result from the Azure CLI: launching Azure CLI: exec: "az": executable file not found in $PATH.
```

Reply:
Oh I didn't notice the duplication. May be I can append D and B with the names?

Reply:
Actually, I'm thinking about something like this: `Standard_D32as_v5 (32 vCPU, 128 GiB RAM)`. WDYT? People could easily find their setup if they decide to look up at [price comparison pages](https://azureprice.net/).

Reply:
I agree.

Reply:
done